### PR TITLE
System.IO files cleanup

### PR DESCRIPTION
**Pull request by adamsitnik**
For the description please refer to commit messages.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 1]
```diff
@@ -731,5 +774,176 @@ private static void Validate(string path, Encoding encoding)
             if (path.Length == 0)
                 throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
         }
+
+        private static byte[] ReadAllBytesUnknownLength(FileStream fs)
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:778]: Did anything change in these methods, or you just moved them between files?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/System.Private.CoreLib/src/System/IO/File.cs:778]: I just moved them and I plan to send a separate PR to change them from using `FileStream` to `RandomAccess`

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/IO/File.cs:778]: Cool

**Comment by EgorBo** [comment]
Probably improved https://github.com/dotnet/perf-autofiling-issues/issues/2375 (windows-x64)

**Pull request merged**